File: urbanairship-layout/src/main/java/com/urbanairship/android/layout/property/WindowSize.java
/* Copyright Airship and Contributors */

package com.urbanairship.android.layout.property;

import com.urbanairship.json.JsonException;

import java.util.Locale;

import androidx.annotation.NonNull;

public enum WindowSize {
    SMALL("small"),
    MEDIUM("medium"),
    LARGE("large");

    @NonNull
    private final String value;

    WindowSize(@NonNull String value) {
        this.value = value;
    }

    @NonNull
    public static WindowSize from(@NonNull String value) throws JsonException {
        for (WindowSize v : WindowSize.values()) {
            if (v.value.equals(value.toLowerCase(Locale.ROOT))) {
                return v;
            }
        }
        throw new JsonException("Unknown WindowSize value: " + value);
    }

    @NonNull
    @Override
    public String toString() {
        return name().toLowerCase(Locale.ROOT);
    }
}

File: urbanairship-layout/src/main/java/com/urbanairship/android/layout/property/WindowSize.kt
/* Copyright Airship and Contributors */
package com.urbanairship.android.layout.property

import com.urbanairship.json.JsonException
import com.urbanairship.json.JsonValue

public enum class WindowSize(
    private val value: String
) {
    SMALL("small"),
    MEDIUM("medium"),
    LARGE("large");

    override fun toString(): String {
        return name.lowercase()
    }

    public companion object {

        @Throws(JsonException::class)
        public fun from(json: JsonValue): WindowSize {
            val value = json.requireString().lowercase()

            return entries.firstOrNull { it.value == value }
                ?: throw JsonException("Unknown WindowSize value: $value")
        }
    }
}
